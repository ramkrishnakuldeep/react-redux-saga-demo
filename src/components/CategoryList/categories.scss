.categories {
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    background: beige;
    padding: 10px;
    position: fixed;
    width: 100%;
    top: 60px;
    z-index: 3;

    .category {
        cursor: pointer;
    }

    .active {
        color: rgb(221, 20, 20);
    }

    .inactive {
        color: gray;
    }
}