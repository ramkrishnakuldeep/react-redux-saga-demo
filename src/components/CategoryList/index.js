import React from 'react';
import './categories.scss';
import { CategoryProps } from '../../propTypes';

const CategoryList = (props) => {
    const {
        categories,
        filterBy,
        filterCategory
    } = props;

    const onCategoryClick = (id) => () => filterCategory(parseInt(id, 10));

    return (
        <div className="categories">
            {
                categories.map((category) => {
                    return (
                        <div className={`category ${filterBy === parseInt(category.id, 10) ? 'active' : 'inactive'}`}
                            onClick={onCategoryClick(category.id)}
                            key={category.id}>
                            {category.name}
                        </div>
                    );
                })
            }
        </div>
    );
};

CategoryList.propTypes = CategoryProps;

export default CategoryList;
