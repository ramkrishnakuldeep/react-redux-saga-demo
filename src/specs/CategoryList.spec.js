import React from 'react';
import { mount } from '../enzyme';
import CategoryList from '../components/CategoryList';
import { MockCategoryList } from './__mock__/data';

const filterCategory = jest.fn();

const props = {
    categories: MockCategoryList.data,
    filterBy: 2,
    filterCategory
};

describe('CategoryList component', () => {
    it('initial render', () => {
        const wrapper = mount(<CategoryList {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should able to dispatch action on click', () => {
        const wrapper = mount(<CategoryList {...props} />);
        wrapper.find('.category').at(1).simulate('click');
        expect(filterCategory).toHaveBeenCalledTimes(1);
    });
});
