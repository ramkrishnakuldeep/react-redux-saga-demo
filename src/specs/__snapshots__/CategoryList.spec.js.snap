// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList component initial render 1`] = `
<CategoryList
  categories={
    Array [
      Object {
        "id": "1",
        "name": "πüÖπü╣πüª",
      },
      Object {
        "id": "2",
        "name": "πâ¼πâçπéúπâ╝πé╣",
      },
      Object {
        "id": "3",
        "name": "πâÖπâôπâ╝πâ╗πé¡πââπé║",
      },
      Object {
        "id": "4",
        "name": "πé¿πâ│πé┐πâí",
      },
      Object {
        "id": "5",
        "name": "πâíπâ│πé║",
      },
    ]
  }
  filterBy={2}
  filterCategory={[MockFunction]}
>
  <div
    className="categories"
  >
    <div
      className="category inactive"
      key="1"
      onClick={[Function]}
    >
      πüÖπü╣πüª
    </div>
    <div
      className="category active"
      key="2"
      onClick={[Function]}
    >
      πâ¼πâçπéúπâ╝πé╣
    </div>
    <div
      className="category inactive"
      key="3"
      onClick={[Function]}
    >
      πâÖπâôπâ╝πâ╗πé¡πââπé║
    </div>
    <div
      className="category inactive"
      key="4"
      onClick={[Function]}
    >
      πé¿πâ│πé┐πâí
    </div>
    <div
      className="category inactive"
      key="5"
      onClick={[Function]}
    >
      πâíπâ│πé║
    </div>
  </div>
</CategoryList>
`;
